src/pages/Game/hooks/useFullscreen.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useState } from 'react';
+
+export const useFullscreen = (element: Element | null) => {
+  const [isFullscreen, setIsFullscreen] = useState(false);
+
+  const onFullScreenExit = useCallback(() => {
+    document.removeEventListener('fullscreenchange', onFullScreenExit);
```

Comment:
❗
А зачем удалять обработчики, а потом снова вешать? Можно [проверить через `document.fullscreenElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/fullscreenchange_event#examples) есть ли после события фулскрин или нет. Получится что-то такое
```suggestion
    setIsFullscreen(Boolean(document.fullscreenElement));
```

Сам обработчик повесится и снимется в `useEffect`

Reply:
Да, действительно. Упростил логику. Только оставил проверку, что именно этот элемент зафулскринен.